File: android_client/app/src/main/java/com/example/duriannet/services/detector/hub_dto/DetectionResultDto.kt
package com.example.duriannet.services.detector.hub_dto

import com.example.duriannet.models.DetectionResult
import com.fasterxml.jackson.annotation.JsonProperty


data class DetectionResultDto(
    @JsonProperty("Label") val label: String,
    @JsonProperty("Confidence") val confidence: Double,
    @JsonProperty("Top") val top: Int,
    @JsonProperty("Left") val left: Int,
    @JsonProperty("Width") val width: Int,
    @JsonProperty("Height") val height: Int,
)

fun DetectionResultDto.toDetectionResult() = DetectionResult(
    label = label,
    confidence = confidence,
    top = top,
    left = left,
    width = width,
    height = height,
)

File: aspnet_server/DurianNet/Hubs/Dto/DetectionResultDto.cs
using MessagePack;

namespace DurianNet.Hubs.Dto
{

    public class DetectionResultDto
    {
        public required string Label { get; set; }
        public double Confidence { get; set; }
        public int Top { get; set; }
        public int Left { get; set; }
        public int Width { get; set; }
        public int Height { get; set; }
    }
}
